.github/workflows/ci-cd.yml

```diff
@@ -123,3 +125,41 @@ jobs:
           POETRY_PYPI_TOKEN_PYPI: ${{ secrets.POETRY_PYPI_TOKEN_PYPI }}
         run: |
           poetry publish
+
+      - name: Get tq42sdk version
+        id: sdk_version
+        run: echo "version=$(poetry version --short)" >> $GITHUB_OUTPUT
+
+      - name: Build changelog
+        id: build_changelog
+        uses: mikepenz/release-changelog-builder-action@v5
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          toTag: ${{ github.sha }}
+          configurationJson: |
+            {
+              "template": "## Release ${{ steps.sdk_version.outputs.version }}\n\n#{{CHANGELOG}}",
+              "categories": [
+                {
+                    "title": "### Changes",
+                    "labels": []
+                }
+              ]
+            }
+
+      - name: Create tag with version
+        continue-on-error: true
```

Comment:
Added this so in case we need to run this pipeline multiple times, the tag might already exist. The release action can execute multiple times.